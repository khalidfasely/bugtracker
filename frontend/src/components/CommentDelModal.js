import Modal from "react-modal";

const CommentDelModal = ({ delModalOpen, setDelModalOpen, deleteComment }) => (
    <Modal
      isOpen={delModalOpen}
      onRequestClose={() => setDelModalOpen(false)}
    >
        Alright <button onClick={deleteComment} data-testid='delete_button'>Delete</button>
    </Modal>
);

export default CommentDelModal;