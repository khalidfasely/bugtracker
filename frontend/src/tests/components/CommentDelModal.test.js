import React from 'react';

import { render, fireEvent } from '@testing-library/react';

import CommentDelModal from '../../components/CommentDelModal';

let props;

beforeEach(() => {
    props = {
        delModalOpen: true,
        setDelModalOpen: jest.fn(),
        deleteComment: jest.fn()
    }
});

test('Should render CommentDelModal correctly', () => {
    const { asFragment } = render(<CommentDelModal {...props} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should delete button fire correctly', () => {
    const { getByTestId } = render(<CommentDelModal {...props} />);
    const buttonDelEl = getByTestId('delete_button');

    expect(props.deleteComment).toHaveBeenCalledTimes(0);

    fireEvent.click(buttonDelEl);

    expect(props.deleteComment).toHaveBeenCalledTimes(1);
});