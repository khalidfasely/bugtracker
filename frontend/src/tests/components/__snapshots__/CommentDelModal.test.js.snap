// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CommentDelModal correctly 1`] = `
<DocumentFragment>
  Alright 
  <button
    data-testid="delete_button"
  >
    Delete
  </button>
</DocumentFragment>
`;
